### ISSUE TITLE
InlineControlStructure CBF issue while adding braces to an if thats returning a nested function

### ISSUE THREAD

**Issue by hchow77** [squizlabs/PHP_CodeSniffer#1590]
Not very happy with how this code came into being, but does seem to have resulted in a pretty bad CBF fix while using Generic.ControlStructures.InlineControlStructure.NotAllowed. Can't really put it into words because it's so hairy, but here's what it starts out as:
```
class Clazz {
    public function foo() {
        $num = 0;
        switch ($num) {
            case 0:
                if (1 > $num)
                    return bar(
                        baz(
                            "foobarbaz"
                        )
                    );
                break;

            default:
                return bar("foobarbaz");
        }
    }
}

```

After running I get:

```
class Clazz {
    public function foo() {
        $num = 0;
        switch ($num) {
            case 0:
                if (1 > $num) {
                    return;
                } bar(
                        baz(
                            "foobarbaz"
                        )
                    );
                break;

            default:
                return bar("foobarbaz");
        }
    }
}
```

Note, it's no longer returning the result of bar, which seems pretty bad. Given the code was awful to begin with, and noticed that making fine adjustments that would have made this code better makes it work right, but I'm surprised that CBF is sneaking in this semicolon in this case in particular.

Appreciate the consideration!

**Comment by gsherwood** [comment]
It looks like the tokenizer thinks the return keyword is the closing statement of the case. So the inline control structure sniff thinks the return keyword is where the if statement should end as well.

So I think the problem is inside the tokenizer itself, although I'm not really sure how to fix it. Inline control structures seem to break everything, all the time :)

**Comment by gsherwood** [comment]
The reason this was failing is because the tokenizer was stopping its search for the IF opener after 3 lines. I've got other code blocks that should stop that search early when they find a semicolon or another scope opener, but neither was found in this specific case.

I've increased the search to 30 lines instead of making it unlimited just because it was originally introduced for extreme edge cases that caused bad performance. I may consider removing it in the future all together.

Thanks for the report.
